packages/collaboration-manager/src/CollaborationManager.ts

```diff
@@ -73,6 +130,10 @@ export class CollaborationManager {
       default:
         console.error('Unknown event type', e);
     }
-    console.log('operation', operation);
+
+    if (operation !== null) {
+      this.#undoRedoManager.putToUndoStack(operation);
+      this.#undoRedoManager.flushRedoStack();
+    }
```

Comment:
This logic also should be inside of undoredo manager

There should be just one method like `push` which would
1. Put operation to undo stack
2. Flush redo stack